File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        left,currsum=0,0
        minlen=float("inf")
        n=len(nums)
        for right in range(n):
            currsum+=nums[right]
            while(currsum>=target):
                if(right-left+1<minlen):
                    minlen=right-left+1
                currsum-=nums[left]
                left+=1
        if(minlen==float("inf")):
            return 0
        else:
            return minlen                

           


        

File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int minlen=1234567;
        int left=0;
        int currsum=0;
        for (int i=0;i<nums.length;i++){
            currsum=currsum+nums[i];
            while(currsum>=target){
                if(i-left+1<minlen){
                    minlen=i-left+1;
                }
                currsum=currsum-nums[left];
                left=left+1;
            }
        }
        if(minlen!=1234567){
                return minlen;
            }
            else{
            return 0;
            }
    }
}